modules/planning/common/frame_open_space.cc

```diff
@@ -233,5 +240,115 @@ const std::vector<const Obstacle *> FrameOpenSpace::obstacles() const {
   return obstacles_.Items();
 }
 
+bool FrameOpenSpace::VPresentationObstacle() {
+  obstacles_num_ = obstacles_.Items().size();
+  if (obstacles_num_ == 0) {
+    AINFO << "no obstacle by perception";
+    return false;
+  }
+  obstacles_vertices_num_ = 4 * Eigen::MatrixXd::Ones(obstacles_num_, 1);
+  for (const auto &obstacle : obstacles_.Items()) {
+    Box2d obstacle_box = obstacle->PerceptionBoundingBox();
+    std::vector<Vec2d> vertices_ccw = obstacle_box.GetAllCorners();
+    std::vector<Vec2d> vertices_cw;
+    while (!vertices_ccw.empty()) {
+      vertices_cw.emplace_back(vertices_ccw.back());
+      vertices_ccw.pop_back();
+    }
+    // As the obstacle is a closed convex set, the first vertice is repeated at
+    // the end of the vector to help transform all four edges to inequality
+    // constraint
+    vertices_cw.push_back(vertices_cw.front());
+    obstacles_vertices_vec_.emplace_back(vertices_cw);
+  }
+  return true;
+}
+
+bool FrameOpenSpace::HPresentationObstacle() {
+  obstacles_A_ = Eigen::MatrixXd::Zero(obstacles_vertices_num_.sum(), 2);
+  obstacles_b_ = Eigen::MatrixXd::Zero(obstacles_vertices_num_.sum(), 1);
+  // vertices using H-represetntation
+  if (!ObsHRep(obstacles_num_, obstacles_vertices_num_, obstacles_vertices_vec_,
+               &obstacles_A_, &obstacles_b_)) {
+    AINFO << "Fail to present obstacle in hyperplane";
+    return false;
+  }
+  return true;
+}
+
+bool FrameOpenSpace::ObsHRep(
+    const std::size_t &obstacles_num,
+    const Eigen::MatrixXd &obstacles_vertices_num,
+    const std::vector<std::vector<Vec2d>> &obstacles_vertices_vec,
+    Eigen::MatrixXd *A_all, Eigen::MatrixXd *b_all) {
+  if (obstacles_num != obstacles_vertices_vec.size()) {
+    AINFO << "obstacles_num != obstacles_vertices_vec.size()";
+    return false;
+  }
+
+  A_all->resize(obstacles_vertices_num.sum(), 2);
+  b_all->resize(obstacles_vertices_num.sum(), 1);
+
+  int counter = 0;
+
+  // start building H representation
+  for (std::size_t i = 0; i < obstacles_num; ++i) {
+    std::size_t current_vertice_num = obstacles_vertices_num(i, 0);
+    Eigen::MatrixXd A_i(current_vertice_num, 2);
+    Eigen::MatrixXd b_i(current_vertice_num, 1);
+
+    // take two subsequent vertices, and computer hyperplane
+    for (std::size_t j = 0; j < current_vertice_num; ++j) {
+      Vec2d v1 = obstacles_vertices_vec[i][j];
+      Vec2d v2 = obstacles_vertices_vec[i][j + 1];
+
+      Eigen::MatrixXd A_tmp(2, 1), b_tmp(1, 1), ab(2, 1);
+      // find hyperplane passing through v1 and v2
+      if (v1.x() == v2.x()) {
```

Comment:
double compare: add a epsilon for eq.

Reply:
found out that if I give it a Epsilon, it will fail my own test with a big margin, which involve an extreme case(vertical and horizontal).

Still prefer ==, cause I didn't formulate the line as y= ax +b, which will have big problem when the line is about to be vertical or horizontal. I formulate it as a1*y = a2*x + b. So "==", won't cause that problem